gst/nnstreamer/tensor_query/tensor_query_client.c

```diff
@@ -478,6 +481,36 @@ _nns_edge_event_cb (nns_edge_event_h event_h, void *user_data)
   return ret;
 }
 
+/**
+ * @brief Internal function to create edge handle.
+ */
+static gboolean
+gst_tensor_query_client_create_edge_handle (GstTensorQueryClient * self)
+{
+  int ret;
+
+  if (!self->edge_h) {
+    ret = nns_edge_create_handle ("TEMP_ID", self->connect_type,
+        NNS_EDGE_FLAG_RECV | NNS_EDGE_FLAG_SEND, &self->edge_h);
+    if (ret != NNS_EDGE_ERROR_NONE)
+      return FALSE;
+
+    nns_edge_set_event_callback (self->edge_h, _nns_edge_event_cb, self);
+
+    if (self->topic)
+      nns_edge_set_info (self->edge_h, "TOPIC", self->topic);
+    if (self->host)
+      nns_edge_set_info (self->edge_h, "HOST", self->host);
+    if (self->port > 0) {
+      gchar *port = g_strdup_printf ("%u", self->port);
+      nns_edge_set_info (self->edge_h, "PORT", port);
+      g_free (port);
```

Comment:
Is it guaranteed that topic, host, and port will be never changed between init and finalize?

Reply:
I updated code to compare old caps string when received gst-caps event.

@gichan-jang @myungjoo 
Discussion: Now it does not guarantee the change of host and port after starting the pipeline.
(If it is necessary, we need to consider adding state-change callback and closing edge handle if pipeline state is changed from 'pause' to 'ready'.

Reply:
You can forbid changing host/port while it is active. (and make it allowed if there is a need :) )

Reply:
Marked todo to prevent changing the properties while pipeline is running.